build.gradle

```diff
@@ -0,0 +1,198 @@
+import static org.apache.tools.ant.taskdefs.condition.Os.*
+
+buildscript {
+    dependencies {
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.1'
+    }
+}
+
+plugins {
+    id 'com.github.ksoichiro.console.reporter' version '0.5.0'
+}
+
+apply plugin: 'java'
+apply plugin: 'jacoco'
+apply plugin: 'signing'
+apply plugin: 'maven'
+apply plugin: 'idea'
+apply plugin: 'com.google.protobuf'
+
+group 'io.leonis'
+version '0.0.1'
+archivesBaseName = "subra-protocol"
+rootProject.description = 'protocols for subra'
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+if (!isFamily(FAMILY_WINDOWS) && !isFamily(FAMILY_MAC) && !isFamily(FAMILY_UNIX))
+    throw new GradleException("Unsupported operating system!")
+
+javadoc {
+    source = sourceSets.main.allJava
+    classpath = configurations.compile
+}
+
+protobuf {
+    protoc {
+        // The artifact spec for the Protobuf Compiler
+        artifact = 'com.google.protobuf:protoc:3.3.0'
+    }
+    generatedFilesBaseDir = "$projectDir/gen"
+}
+
+clean {
+    delete protobuf.generatedFilesBaseDir
+}
+
+idea {
+    module {
+        sourceDirs += file("${protobuf.generatedFilesBaseDir}/main/java")
+    }
+}
+
+compileJava {
+    options.compilerArgs << '-parameters'
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    // tests
+    testCompile 'org.testng:testng:6.11'
+
+    compile 'com.google.protobuf:protobuf-java:3.3.0'
+
+    protobuf fileTree('src/main/protobuf/')
+}
+
+test {
+    useTestNG()
+    testLogging {
+        events "PASSED", "STARTED", "FAILED", "SKIPPED"
+        showStandardStreams = true
+    }
+}
+
+tasks.withType(Javadoc) {
```

Comment:
This can be removed since there's no java source.

Reply:
you can't publish jars to mavencentral without a javadoc jar

Reply:
That task is not for building javadoc jars but for validating there are no errors.